tests/test_nwm22_grid_loading.py

```diff
@@ -0,0 +1,65 @@
+
+from pathlib import Path
+import filecmp
+
+import pandas as pd
+
+from teehr.loading.nwm22.nwm_grid_data import fetch_and_format_nwm_grids
+from teehr.loading.nwm_common.utils_nwm import build_zarr_references
+from teehr.loading.nwm22.const_nwm import (
+    NWM22_UNIT_LOOKUP,
+)
+
+TEST_DIR = Path("tests", "data", "nwm22")
+TEMP_DIR = Path("tests", "data", "temp")
+WEIGHTS_FILEPATH = Path(TEST_DIR, "onehuc10_weights.parquet")
+
+
+def test_grid_zarr_reference_file():
+
+    component_paths = [
+        "gcs://national-water-model/nwm.20201218/forcing_analysis_assim/nwm.t00z.analysis_assim.forcing.tm00.conus.nc" # noqa
+    ]
+
+    _ = build_zarr_references(
+        remote_paths=component_paths,
+        json_dir=TEMP_DIR,
+        ignore_missing_file=False
+    )
+
+    json_file = Path(TEMP_DIR,
+                     "nwm.20201218.nwm.t00z.analysis_assim.forcing.tm00.conus.nc.json") # noqa
+    test_file = Path(TEST_DIR, "grid_benchmark.json")
+
+    assert filecmp.cmp(test_file, json_file, shallow=False)
+
+
+def test_grid_fetch_and_format():
```

Comment:
@samlamont you think it would make sense to reference the json file in TEST_DIR instead so it does not rely on the results from a previous test to pass?

Reply:
Yes good point, we could move a copy of `nwm.20201218.nwm.t00z.analysis_assim.forcing.tm00.conus.nc.json` into the TEST_DIR and access the file there

Reply:
couldn't we just use the one we are comparing to instead?

Reply:
well `fetch_and_format` gets parameters from the filename itself, like the day and z-hour of the file to access from GCS https://github.com/RTIInternational/teehr/blob/23daf1fbab6c6c957c1e6110425c53656cdbbf60/src/teehr/loading/nwm_common/grid_utils.py#L96

Reply:
ah, I see.  i guess you could just use the correct file name since it is in a separate directory from the one where the test will write.